### PR TITLE
16735: Submit Button Validations

### PR DESCRIPTION
**Pull request by psk86**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [16735](https://devops.dchbx.org/redmine/issues/16735) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @DFos85 |

### CODE REVIEW

File: app/controllers/insured/family_members_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def index
     @selectedTab = "householdInfo"
     @allTabs = NavigationHelper::getAllTabs
     @allTabs << {"title" => "Review and Submit", "id" => "reviewAndSubmit"}
-    @continueEnabled = (@family.applications.length > 0) ? @family.applications[0].financial_application_complete? : true 
+    #@continueEnabled = (@family.applications.length > 0) ? @family.applications[0].financial_application_complete? : true # what? cant randomly check the first application ?? make use of application_in_progress method
```
Comment (billy-ran-away) [inline app/controllers/insured/family_members_controller.rb]: Good catch!

File: app/models/financial_assistance/application.rb [hunk 2]
```diff
@@ -305,22 +318,84 @@ def eligibility_determinations_for_year(year)
...
+  def application_submission_validity
+    # Mandatory Fields before submission
+    validates_presence_of :hbx_id, :applicant_kind, :request_kind, :motivation_kind, :us_state, :is_ridp_verified
+    # User must agree with terms of service check boxes
+    validates_acceptance_of :medicaid_terms, :attestation_terms, :submission_terms, :medicaid_insurance_collection_terms, :report_change_terms, :parent_living_out_of_home_terms, accept: true
   end
 
   def is_application_valid?
+    #self.save!(context: :submission)
+    self.valid?(:submission) ? true : false
```
Comment (billy-ran-away) [inline app/models/financial_assistance/application.rb]: I like the use of the context option for the validations, good job @psk86!

File: spec/models/financial_assistance/application_spec.rb [hunk 3]
```diff
@@ -112,5 +113,45 @@
         expect(ed).to eq eligibility_determination1
       end
     end
+
+    context "check the validity of an application" do
```
Comment (billy-ran-away) [inline spec/models/financial_assistance/application_spec.rb]: Good specs!

**Comment by Kevin-Wei** [comment]
rebased squashed into data-model-new

**Pull request closed without merging**